src/views/home/components/transaction-form/transaction-form.view.tsx

```diff
@@ -12,19 +12,32 @@ import Icon from "src/views/shared/icon/icon.view";
 import List from "src/views/home/components/list/list.view";
 import Button from "src/views/shared/button/button.view";
 import AmountInput from "src/views/home/components/amount-input/amount-input.view";
-import { Chain, TransactionData } from "src/domain";
+import { Chain, Token, TransactionData } from "src/domain";
 import { useEnvContext } from "src/contexts/env.context";
+import { AsyncTask } from "src/utils/types";
+import { getBalance } from "src/adapters/ethereum";
 
 interface TransactionFormProps {
   onSubmit: (transactionData: TransactionData) => void;
+  transaction?: TransactionData;
+  account: AsyncTask<string, string>;
```

Comment:
I'm not sure. It's true that if we don't have `account.data` you can't continue because balance is zero but I think that we should render something and we can render the whole component without `account.data`